pkg/runtest/run_test.go

```diff
@@ -34,6 +34,11 @@ func Test(t *testing.T) {
 			continue
 		}
 		sysTarget1 := targets.Get(sysTarget.OS, sysTarget.Arch)
+		if runtime.GOOS == "freebsd" && sysTarget1.PtrSize == 4 {
+			// The default DataOffset collides with a runtime mapping
```

Comment:
What does this mean? If simple tests are not working, then all fuzzing should be broken as well?

Reply:
We adjust DataOffset for freebsd/386, but I didn't see a good way to do the same for test/32_shmem or test/32_fork_shmem when running on FreeBSD.

Reply:
Ah, I see, this is test OS running on freebsd. That's not as concerning :)

Perhaps we could move more of real host OS properties to test OS in sys/targets/targets.go here:
``` 
	for _, target := range List["test"] {
		if List[goos] != nil {
			if host := List[goos][runtime.GOARCH]; host != nil {
				target.CCompiler = host.CCompiler
				target.CPP = host.CPP
				if goos == "freebsd" {
					// For some configurations -no-pie is passed to the compiler,
					// which is not used by clang.
					// Ensure clang does not complain about it.
					target.CFlags = append(target.CFlags, "-Wno-unused-command-line-argument")
					// When building executor for the test OS, clang needs
					// to link against the libc++ library.
					target.CFlags = append(target.CFlags, "-lc++")
				}
			}
		}
		target.BuildOS = goos
	}
```

Reply:
It's not quite so simple: it's the freebsd/386 target that has a special constraint, not freebsd/amd64, so I need to map the 32-bit test targets to the freebsd/386 target, not that of the host system (an amd64 FreeBSD workstation in my case). I will try to come up with something here though.

Reply:
See PR 2038.